src/python/dbsClient/apis/dbsClient.py

```diff
@@ -10,6 +10,157 @@
 import sys
 import urllib.request, urllib.parse, urllib.error
 
+def parseStream(results):
+    """"
+    Parse given stream of results
+
+    :param: results: list JSON records
+    :type: list
+    :return: generator of JSON records
+    """
+    for rec in results.split('\n'):
+        if rec:
+            yield json.loads(rec)
+
+def aggAttribute(results, attr):
+    """
+    performs aggregation based on given attribute
+
+    :param: results: list JSON records
+    :type: list
+    :param: attr is name of the attribute in JSON record, e.g. run_num
+    :type: string
+    :return: aggregated list of JSON records
+    """
+    rows = []
+    for row in results:
+        if attr in row and not isinstance(row[attr], list):
```

Comment:
I do not know what exactly is the use case here. However, I wonder why a list data type is skipped here? Perhaps we should instead `extend` the `rows` variable?